scripts/js/commands/checkStaleImages.ts

```diff
@@ -12,42 +12,60 @@
 
 import { $ } from "zx";
 import { globby } from "globby";
-import { zip } from "lodash-es";
 
 import { zxMain } from "../lib/zx.js";
 
+// Files that are used in closed source, so should not be removed.
+// Format: full file path, e.g. "public/images/guides/paulibasis.png"
+const ALLOW_LIST = new Set<string>([]);
+
 zxMain(async () => {
   const paths = await getStrippedImagePaths();
-  const results = await Promise.all(paths.map((fp) => isImageUnused(fp)));
-  const unused = zip(paths, results)
-    .filter(([_fp, result]) => result)
-    .map(([fp]) => `public/${fp}`);
-  if (unused.length === 0) {
+  let allGood = true;
+  let numFilesChecked = 0;
+  for (const fp of paths) {
```

Comment:
The `await Promise.all` approach should work fine here. There won't be a race condition with `numFilesChecked` since Node is single-threaded. I'd expect it to make a difference since there's a lot of time waiting for commands to run. Just move L27 -> L34 so there's no `await` between incrementing and logging.

Reply:
I don't think this would work well because we could trigger too many concurrent calls to ripgrep, which can peg the system.